File: Models/DTAR/DT_StepDetail.cs
namespace DTARServer.Models;

[System.Serializable]
public class DT_StepDetail : DT_Hero
{
	public int itemNumber;

	public DT_QualityAssurance check;

#if !UNITY
	public DT_StepDetail() : base()
    {
    }

	public override List<DT_Document> CollectDocuments(List<DT_Document> list)
	{
		base.CollectDocuments(list);

		return list;
	}

	public DT_StepDetail ShallowCopy()
	{
		var result = (DT_StepDetail)this.MemberwiseClone();
		result.assetReferences = null;
		result.ClearKeys();
		result.DeReference();
		return result;
	}

#endif
}


[System.Serializable]
public class DT_Note : DT_StepDetail
{

}

[System.Serializable]
public class DT_Caution : DT_StepDetail
{
}

[System.Serializable]
public class DT_Warning : DT_StepDetail
{
}

[System.Serializable]
public class DT_StepAction : DT_StepDetail
{
}

[System.Serializable]
public class DT_CAD : DT_StepDetail
{
}

[System.Serializable]
public class DT_Media : DT_StepDetail
{
}

File: Models/DTAR/dT_StepDetail.ts
export interface DT_StepDetail extends DT_Hero {
	itemNumber: number;
	check: DT_QualityAssurance;
}

export interface DT_Note extends DT_StepDetail {

}

export interface DT_Caution extends DT_StepDetail {

}

export interface DT_Warning extends DT_StepDetail {

}

export interface DT_StepAction extends DT_StepDetail {

}

export interface DT_CAD extends DT_StepDetail {

}

export interface DT_Media extends DT_StepDetail {

}